Assets/Scripts/PlayerManager.cs

```diff
@@ -1,25 +1,30 @@
+using System;
 using UnityEngine;
+using UnityEngine.Events;
 using UnityEngine.UI;
 
 public class PlayerManager : MonoBehaviour
 {
     [SerializeField] private Button testButton;
+    Player player;
 
-
-    private void Start()
+    public void Init(string prefabName)
     {
-        AssetsDatabase assetsDatabase = new AssetsDatabase();
-        SpawnPlayer();
+       SpawnPlayer(prefabName);
     }
 
-    public void SpawnPlayer()
+    private void SpawnPlayer(string prefabName)
```

Comment:
That's a personal preference, but I usually avoid search things from their name, maybe some sort of ENUM to handle it?